File: prediction/price_prediction.py
from flask import Flask,request,jsonify;
import pickle
import numpy as np

app=Flask(__name__)

# Load your trained model
with open('model.pkl', 'rb') as model_file:
    regressor = pickle.load(model_file)

# Function to predict price based on input data
def predict_price(Day,Month,Year):
    try:
       
        
        input_data =(Day,Month,Year)

        # Predict the price for a new input
        predicted_price = regressor.predict(input_data)

        # Return the predicted price
        return predicted_price[0]  # Assuming the prediction is a single value
    except Exception as e:
        return str(e)

if __name__ == "__main__":
    import sys

    if len(sys.argv) != 3:
        print("Usage: python price_prediction.py <vegetable_name> <date>")
        sys.exit(1)

    vegetable = sys.argv[1]
    date = sys.argv[2]
    prediction = predict_price()
    print("Predicted price:", prediction)

File: frontend/component/PricePrediction/PricePrediction.js
import React, { useState } from "react";
import axios from "axios";

function PricePrediction() {
  const [predictedPrice, setPredictedPrice] = useState(null);
  const [inputData, setInputData] = useState({
    Day: "",
    Month: "",
    Year:"",
  });

  const predictPrice = async () => {
    try {
      const response = await axios.post("/predict-price", inputData);
      setPredictedPrice(response.data.predictedPrice);
    } catch (error) {
      console.error("Error:", error);
    }
  };

  return (
    <div>
      <h2>Vegetable Price Prediction</h2>
      <label>
       Day:
        <input
          type="text"
          value={inputData.Day}
          onChange={(e) => setInputData({ ...inputData, Day: e.target.value })}
        />
      </label>
      <br />
      <label>
        Month:
        <input
          type="text"
          value={inputData.Month}
          onChange={(e) => setInputData({ ...inputData, Month: e.target.value })}
        />
      </label>
      <label>
       Year:
        <input
          type="text"
          value={inputData.Year}
          onChange={(e) => setInputData({ ...inputData, Year: e.target.value })}
        />
      </label>
      <br />
      <button onClick={predictPrice}>Predict Price</button>
      <br />
      {predictedPrice !== null && <p>Predicted Price: Rs {predictedPrice}</p>}
    </div>
  );
}

export default PricePrediction;
